src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_Border.cs

```diff
@@ -615,6 +615,52 @@ public async Task Nested_Element_Tapped()
 			Assert.AreEqual(1, outerBorderRightTaps);
 		}
 
+		[TestMethod]
+		[RunsOnUIThread]
+		public async Task When_Child_Set_Same_Reference()
+		{
+			var SUT = new Border() { Width = 100, Height = 100 };
+			WindowHelper.WindowContent = SUT;
+			await WindowHelper.WaitForLoaded(SUT);
+
+			var child = new MeasureArrangeCounterButton();
+			SUT.Child = child;
+
+			await WindowHelper.WaitFor(() => child.ArrangeCount > 0);
+
+			Assert.IsTrue(child.MeasureCount > 0);
+			Assert.IsTrue(child.ArrangeCount > 0);
+
+			var oldMeasureCount = child.MeasureCount;
+			var oldArrangeCount = child.ArrangeCount;
+
+			SUT.Child = child;
+
+			await WindowHelper.WaitFor(() => child.MeasureCount > oldMeasureCount);
+
+			Assert.IsTrue(child.MeasureCount > oldMeasureCount);
+			Assert.IsTrue(child.ArrangeCount > oldArrangeCount);
```

Comment:
Same here, if we can assert against the exact value

Reply:
Same https://github.com/unoplatform/uno/issues/17861